syntax = 'proto3';
import "google/protobuf/any.proto";

import "proto/geometry.proto";

package TbotsProto;

message Path
{
    repeated Point point = 1;
}

message PathVisualization
{
    repeated Path path = 1;
}

message Obstacles
{
    repeated Polygon polygon = 1;
    repeated Circle circle   = 2;
}

message PassWithRating
{
    double rating = 1;
    Pass pass_    = 2;  // needs the _ because pass is a keyword in python
}

message Pass
{
    Point passer   = 1;
    Point receiver = 2;
}

message PassVisualization
{
    repeated PassWithRating best_passes = 1;
}
